drivers/cmd/gcloud-pubsub/main.go

```diff
@@ -0,0 +1,168 @@
+		client, err = pubsub.NewClient(context.Background(), project, clientOption)
+	} else {
+		client, err = pubsub.NewClient(context.Background(), project)
+	}
+	if err != nil {
+		panic(errors.New("unable to create gcloud pubsub client"))
+	}
+	return client
+}
+
+func loadOptions() {
+	events, ok := driver.GetOption("dynamicData.collapsedEvents")
+	if !ok {
+		dipper.Logger.Panicf("[%s] no pubsub subscription defined for pubsub driver", driver.Service)
+	}
+	pubsubEvents, ok := events.(map[string]interface{})
+	if !ok {
+		dipper.Logger.Panicf("[%s] pubsub subscription data should be a map of event to conditions", driver.Service)
+	}
+
+	for _, pubsubEvent := range pubsubEvents {
+		for _, condition := range pubsubEvent.([]interface{}) {
+			project, ok := dipper.GetMapDataStr(condition, "project")
+			if !ok {
+				dipper.Logger.Warningf("[%s] failed to get pubsub project in condition", driver.Service)
+				continue
+			}
+			subscriptionName, ok := dipper.GetMapDataStr(condition, "subscriptionName")
+			if !ok {
+				dipper.Logger.Warningf("[%s] failed to get pubsub subscription name in condition", driver.Service)
+				continue
+			}
+			serviceAccount, ok := dipper.GetMapDataStr(condition, "service_account")
+			if !ok {
+				serviceAccount = ""
+			}
+
+			// TODO: should I also differ subscriber based on rawEvent?
+			subscriberName := project + ":" + subscriptionName + ":" + serviceAccount
+			subscriberConfig, ok := subscriberConfigs[subscriberName]
+			if !ok {
+				subscriberConfigs[subscriberName] = &SubscriberConfig{
+					Project:          project,
+					SubscriptionName: subscriptionName,
+					ServiceAccount:   serviceAccount,
+					Conditions: []map[string]interface{}{
+						condition.(map[string]interface{}),
+					},
+				}
+			} else {
+				subscriberConfig.Conditions = append(
+					subscriberConfig.Conditions,
+					condition.(map[string]interface{}),
+				)
+			}
+		}
+	}
+}
+
+func start(msg *dipper.Message) {
+	loadOptions()
+	go subscribeAll()
+}
+
+func subscribeAll() {
+	for _, subscriberConfig := range subscriberConfigs {
+		go func(config *SubscriberConfig) {
+			for {
+				func() {
+					project := config.Project
+					subscriptionName := config.SubscriptionName
+					serviceAccount := config.ServiceAccount
+
+					defer dipper.SafeExitOnError("[%s] re-subscribing to gcloud pubsub %s", driver.Service, subscriptionName)
+					client := getPubsubClient(serviceAccount, project)
+
+					defer client.Close()
+					sub := client.Subscription(subscriptionName)
+
+					ctx := context.Background()
+					err := sub.Receive(ctx, func(ctx context.Context, msg *pubsub.Message) {
+						data := dipper.DeserializeContent(msg.Data)
+
+						eventData := data.(map[string]interface{})
+
+						// push condition data into payload
+						eventData["project"] = project
+						eventData["subscriptionName"] = subscriptionName
+						eventData["service_account"] = serviceAccount
```

Comment:
instead of injecting the meta into `eventData`, then compare, maybe we can just extract the `condition` and compare only the `condition` to the `data`.